### PR TITLE
feat: balance query

### PR DESCRIPTION
**Pull request by nejcm**

### CODE REVIEW

**Comment by nejcm** [comment]
@ebma Might need you a bit on this just for the query for balance. I prepared the components and hooks and updated the UI already

**Comment by ebma** [comment]
Can you point me to the exact locations in the code where you have TODOs for which you'd require my input?

**Comment by nejcm** [comment]
> Can you point me to the exact locations in the code where you have TODOs for which you'd require my input?

Inside the useBalance hook.

File: src/hooks/useBalance.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    proofSize: '1000000',
+  }) as WeightV2,
+  storageDepositLimit: null,
+});
+
+export const useBalance = (tokenAddress?: string, options?: QueryOptions): UseBalanceResponse => {
+  const {
+    state: { api },
+  } = useNodeInfoState();
+  const { address } = useGlobalState().walletAccount || {};
```
Comment (ebma) [inline src/hooks/useBalance.ts:30]: Maybe, in this case we should actually consider falling back to some hard-coded account address. As far as I know, for querying data the account doesn't have to actually 'exist' on the network or rather, actually have some tokens.

With the current implementation, the user would only see some balances loading once he connects his wallet account. But as a user, I would expect to already be able to interact with the Swap UI on a basic level without needing to connect my account.

Comment (ebma) [reply src/hooks/useBalance.ts:30]: So I would suggest we define a constant 'FALLBACK_QUERY_SUBSTRATE_ACCOUNT' or similar, that is a string holding any account address. 
We could use '5GjwVRkXVyFjZxHfbGWNP3ed44pjvwFes8AJzniMzznWa27i', which is the address of the joint product-tech account.

**Review by ebma — COMMENTED** [review]
Thanks for fixing the theme.

I noticed that since we don't have any swap pools or backstop pool configured, maybe it would make sense to show some small info text instead of showing nothing at all? But that'd be unrelated to the balance query, so it doesn't have to be done here.

Besides that I left some other comments but it looks good to me overall.

File: src/components/Asset/Price/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { memo } from 'preact/compat';
+import { Skeleton } from '../../Skeleton';
+
+// ! TODO
```
[the remark this answers is not in the document]
Comment (gonzamontiel) [reply src/components/Asset/Price/index.tsx:6]: +1

**Review by gonzamontiel — APPROVED** [review]
I now understand that there's still some pending data to fetch, so this PR includes 'mocked' data. If that is correct, the changes make sense (but don't forget to include TODOs where needed, so we can track the hardcoded pieces).

File: src/services/api/assets/index.ts
[the remark this answers is not in the document]
Comment (nejcm) [reply src/services/api/assets/index.ts:1]: Deleted it just from this branch. This was before I saw you are implementing stuff there.

File: src/hooks/useBalance.ts (hunk 1 above)
Comment (nejcm) [reply src/hooks/useBalance.ts:30]: Maybe I am misunderstanding or maybe the name of the hook needs to be more descriptive. This should be used for fetching user/wallet balances (his tokens, deposits...). So maybe renaming the hook?

File: src/components/Asset/Price/index.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  amount?: number;
+  prefix?: ReactNode;
+  options?: any;
+  loader?: ReactNode;
+  fallback?: ReactNode;
+};
+
+const TokenPrice = memo(
+  ({ address, amount = 1, prefix, loader, fallback = null, options }: TokenPriceProps): JSX.Element | null => {
+    const { data, isLoading } = { data: 120.3, isLoading: false }; //useTokenPrice(address, chainId, options);
```
[the remark this answers is not in the document]
Comment (nejcm) [reply src/components/Asset/Price/index.tsx:18]: Yes. Not implemented yet. I added a more descriptive TODO at the start of the component

File: src/hooks/useBalance.ts (hunk 1 above)
Comment (ebma) [reply src/hooks/useBalance.ts:30]: Ah true, it's always just about the user balance. Nevermind then, we wouldn't need the fallback for that.

File: src/services/api/assets/index.ts
[the remark this answers is not in the document]
Comment (ebma) [reply src/services/api/assets/index.ts:1]: Okay. I guess we can re-add it once needed.

**Review by ebma — APPROVED** [review]
LGTM

**Pull request merged**